### PR TITLE
add baguette66 keyboard

### PR DESCRIPTION
**Pull request by moyi4681**
reopen https://github.com/qmk/qmk_firmware/pull/15933

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).
](https://github.com/qmk/qmk_firmware/pull/15933)](https://github.com/qmk/qmk_firmware/pull/15933)](https://github.com/qmk/qmk_firmware/pull/15933)

### CODE REVIEW

**Review by drashna — APPROVED** [review]

**Comment by moyi4681** [comment]
can you guys check and approve this pull request again,  thanks @drashna

File: keyboards/kbdfans/baguette66/soldered/config.h [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+#define MATRIX_COL_PINS { C6, C7, F7, F6, B0, B1, B2, B3, D0, D1, D2, D3, D5, D4, D6}
+#define UNUSED_PINS
+
+/* COL2ROW, ROW2COL*/
+#define DIODE_DIRECTION COL2ROW
+/* Debounce reduces chatter (unintended double-presses) - set 0 if debouncing is not needed */
+#define DEBOUNCE 5
+/* disable these deprecated features by default */
+#define NO_ACTION_MACRO
+#define NO_ACTION_FUNCTION
```
Comment (fauxpark) [inline keyboards/kbdfans/baguette66/soldered/config.h:41]:
```suggestion
```

File: keyboards/kbdfans/baguette66/soldered/config.h [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+#define USB_SUSPEND_WAKEUP_DELAY 5000
```
Comment (fauxpark) [inline keyboards/kbdfans/baguette66/soldered/config.h:42]: Is this needed?

File: keyboards/kbdfans/baguette66/rgb/config.h [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+#define MATRIX_COL_PINS { C6, C7, F7, F6, B0, B1, B2, B3, D0, D1, D2, D3, D5, D4, D6}
+#define UNUSED_PINS
+
+/* COL2ROW, ROW2COL*/
+#define DIODE_DIRECTION COL2ROW
+/* Debounce reduces chatter (unintended double-presses) - set 0 if debouncing is not needed */
+#define DEBOUNCE 5
+/* disable these deprecated features by default */
+#define NO_ACTION_MACRO
+#define NO_ACTION_FUNCTION
```
Comment (fauxpark) [inline keyboards/kbdfans/baguette66/rgb/config.h:41]:
```suggestion
```

File: keyboards/kbdfans/baguette66/rgb/config.h [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+#define USB_SUSPEND_WAKEUP_DELAY 5000
```
Comment (fauxpark) [inline keyboards/kbdfans/baguette66/rgb/config.h:42]: As above.

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by tzarc — APPROVED** [review]

**Pull request merged**